model/access.go

```diff
@@ -74,41 +74,21 @@ func (me *AccessData) IsExpired() bool {
 }
 
 func (ad *AccessData) ToJson() string {
-	b, err := json.Marshal(ad)
-	if err != nil {
-		return ""
-	} else {
-		return string(b)
-	}
+	b, _ := json.Marshal(ad)
+	return string(b)
 }
 
-func AccessDataFromJson(data io.Reader) *AccessData {
-	decoder := json.NewDecoder(data)
-	var ad AccessData
-	err := decoder.Decode(&ad)
-	if err == nil {
-		return &ad
-	} else {
-		return nil
-	}
+func AccessDataFromJson(data io.Reader) (ad *AccessData) {
+	json.NewDecoder(data).Decode(&ad)
+	return
```

Comment:
I'm personally not a fan of naked returns, as they generally force me to reason through another loop on the code: "`return` nothing? oh, `(ad *AccessData)`. When does `ad` get assigned?" Is it worth saving the single line relative to the following? 1/5, given you've already done the work ;P
```
func AccessDataFromJson(data io.Reader) *AccessData {
    var ad AccessData
    json.NewDecoder(data).Decode(&ad)
    return ad
}
```

Reply:
Actually, there's also https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters which says:
>  Don't name result parameters just to avoid declaring a var inside the function; that trades off a minor implementation brevity at the cost of unnecessary API verbosity.

Reply:
It makes very little difference to me. Updated.